apps/wing/src/commands/compile.ts

```diff
@@ -90,8 +94,19 @@ export async function compile(entrypoint: string, options: ICompileOptions) {
 
   const arg = `${normalPath(wingFile)};${normalPath(workDir)}`;
   log(`invoking %s with: "%s"`, WINGC_COMPILE, arg);
-  const compileResult = wingCompiler.invoke(wingc, WINGC_COMPILE, arg);
+  let compileResult;
+  try {
+    compileResult = wingCompiler.invoke(wingc, WINGC_COMPILE, arg);
+  } catch (e) {
+    // This is a bug in Wing, not the user's code.
+    console.error(e);
+    console.log("  " + chalk.bold.red("error:") + " An internal error occurred while Wing was compiling. Please report this bug by creating an issue on GitHub (github.com/winglang/wing) with your Wing code.");
```

Comment:
```suggestion
    console.log("\n\n" + chalk.bold.red("Internal error:") + " An internal compiler error occurred. Please report this bug by creating an issue on GitHub (github.com/winglang/wing/issues) with your code and this trace.");
```

Reply:
We should write a style guide for errors at some point

(related: https://developers.google.com/tech-writing/error-messages)